src/ng/animate.js

```diff
@@ -1,6 +1,117 @@
 'use strict';
 
 var $animateMinErr = minErr('$animate');
+var ELEMENT_NODE = 1;
+
+function mergeClasses(a,b) {
+  if (!a && !b) return '';
+  if (!a) return b;
+  if (!b) return a;
+  if (isArray(a)) a = a.join(' ');
+  if (isArray(b)) b = b.join(' ');
+  return a + ' ' + b;
+}
+
+function extractElementNode(element) {
+  for (var i = 0; i < element.length; i++) {
+    var elm = element[i];
+    if (elm.nodeType === ELEMENT_NODE) {
+      return elm;
+    }
+  }
+}
+
+var $$AnimateClassResolverProvider = [function() {
+  var ADD_CLASS = 1;
+  var REMOVE_CLASS = -1;
+
+  this.$get = [function() {
+    return function(existing, toAdd, toRemove) {
+      var flags = {};
+      existing = splitClasses(existing);
+
+      toAdd = splitClasses(toAdd);
+      forEach(toAdd, function(value, key) {
+        flags[key] = ADD_CLASS;
+      });
+
+      toRemove = splitClasses(toRemove);
+      forEach(toRemove, function(value, key) {
+        if (flags[key] === ADD_CLASS) {
+          delete flags[key];
```

Comment:
I wonder if `delete` slows things down. It doesn't seem to be necessary.
E.g. `flags[key] = flags[key] ? undefined : REMOVE_CLASS` (and then later check for a truthy value in `flags[key]`).